docs/topics/dokka-migration.md

```diff
@@ -438,8 +438,75 @@ DGP v2 now supports Gradle build cache and configuration cache, improving build
 * To enable build cache, follow instructions in the [Gradle build cache documentation](https://docs.gradle.org/current/userguide/build_cache.html#sec:build_cache_enable).
 * To enable configuration cache, follow instructions in the [Gradle configuration cache documentation](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:usage:enable ).
 
+## Troubleshoot your migration
+
+In large projects, Dokka can consume a significant amount of memory to generate documentation. This issue can be due to Gradle's memory allocation limits, 
+and Dokka's processing of large volumes of data, among other factors.
+
+When Dokka generation runs out of memory, the build fails, and Gradle can throw exceptions like `java.lang.OutOfMemoryError: Metaspace`.
+
+Active efforts are underway to improve Dokka's performance, although Gradle limitations exist.
+
+As a workaround, when experiencing memory exceptions, you can try:
+
+* [Increasing heap space](#increase-heap-space)
+* [Running Dokka in-process](#run-dokka-in-process)
+
+### Increase heap space
+
+One way to resolve memory issues is to increase the amount of Java heap memory for the Dokka generator process. 
+In the `build.gradle.kts` file, adjust the 
+following configuration option:
+
+```kotlin
+dokka {
+    // Dokka generates a new process managed by Gradle
+    workerIsolation = ProcessIsolation {
+        // Configures heap size
+        minHeapSize = "4g"
+    }
+}
+```
+
+The example above shows a heap size of `4g`; you need to adjust and test until you find the best solution for your build.
+
+If you have to increase the memory by an unreasonable amount, for example, significantly more than the memory required for Gradle,
+[report an issue on Dokka's GitHub repository](https://kotl.in/dokka-issues).
+
+> You have to apply this configuration to each subproject. It's recommended that you configure Dokka in a convention 
+> plugin applied to all subprojects.
+>
+{style="note"}
+
+### Run Dokka in-process
```

Comment:
how about - Run Dokka in a Gradle process?
Although in-process might work as well - it's a setting I suppose 🤔